### PR TITLE
Adds some default ESLint configs

### PR DESCRIPTION
**Pull request by pmcelhaney**
Needs more testing and I'd like to get some feedback before merging.

### CODE REVIEW

File: lib/rules/rules.json [hunk 1]
```diff
@@ -1,37 +1,43 @@
...
-  "any": "Consider using the native Array.prototype.some()",
-  "isNaN": "Consider using the native isNaN() or ES6 Number.isNaN()",
-  "extendOwn": "Consider using the native ES6 Object.assign()",
-  "assign": "Consider using the native ES6 Object.assign()",
-  "keys": "Consider using the native Object.keys()",
-  "repeat": "Consider using the native ES6 String.prototype.repeat()",
-  "toLower": "Consider using the native String.prototype.toLowerCase()",
-  "toUpper": "Consider using the native String.prototype.toUpperCase()",
-  "trim": "Consider using the native String.prototype.trim()"
+  "concat":      { "compatible": true, "alternative": "Array.prototype.concat()" },
```
Comment (pmcelhaney) [inline lib/rules/rules.json]: We can add more metadata to this file, such as which browsers are supported. Eventually we may even generate the documentation and run automated compatibility tests from the configuration file.

File: package.json [hunk 2]
```diff
@@ -26,12 +26,14 @@
...
   "dependencies": {
-    "kebab-case": "^1.0.0"
+    "kebab-case": "^1.0.0",
+    "lodash": "^4.13.1"
   },
   "devDependencies": {
     "coveralls": "^2.11.9",
     "eslint": "^3.0.0",
     "istanbul": "^0.4.4",
+    "lodash": "^4.13.1",
```
Comment (stevemao) [inline package.json]: duplicated dep?

File: lib/rules/rules.json (hunk 1 above)
Comment (stevemao) [inline lib/rules/rules.json]: Nice! Also note that `_.forEach` works on objects. In such case, I think it shouldn't warn or display a different warning message.

File: index.js [hunk 3]
```diff
@@ -1,3 +1,34 @@
...
+const all = Object.keys(rules);
+const compatible = Object.keys(rules).filter(key => rules[key].compatible);
+const incompatible = Object.keys(rules).filter(key => !rules[key].compatible);
+
+const ERROR = 2;
+const WARN = 1;
+
+module.exports.configs = {
+    'all': {
+      rules: _.zipObject(all, Array(all.length).fill(WARN))
```
Comment (stevemao) [inline index.js]: This doesn't need `_.zipObject()`, `all.map()` should work.

File: package.json (hunk 2 above)
Comment (pmcelhaney) [inline package.json]: Thanks. I'll remove it.

File: index.js (hunk 3 above)
Comment (pmcelhaney) [inline index.js]: I need to return an object. I don't think `map()` will work, but can probably make it happen with `reduce()`. Guess I don't need Lodash!

File: lib/rules/rules.json (hunk 1 above)
Comment (pmcelhaney) [inline lib/rules/rules.json]: Thanks! I like "safe". I think it captures the essence of "yeah, you almost certainly don't need _ here" without getting bogged down in technicalities.

Just to throw some more words out there: substitutable, similar, identical, isomorphic, congruent.

**Pull request closed without merging**